api/src/main/java/org/openmrs/Person.java

```diff
@@ -1093,5 +1093,19 @@ public void setId(Integer id) {
 		setPersonId(id);
 		
 	}
-	
+
+	@JsonIgnore
+	public boolean hasZeroNotVoidedNames() {
```

Comment:
You're right, `hasNoVoidedNames` sounds better 👍 
`hasVoidedName` would miss-communicate the the logic inside this method because given an empty list of names, it should still return `true`. I have written a test for this scenario, `hasZeroNotVoidedNames_whenPersonHasNoNames_shouldBeTrue`

Reply:
In terms of logic everything is ok ✌️ 
The naming is tricky. Originally the condition was:
```java
if (!person.getVoided() && !atLeastOneNonVoidPersonNameLeft) {
   //mark validation error
}
```
When adding a new method, if the only place it's used contains a negation, then I try to reverse its logic. I believe it makes reading code a little bit easier later.
So in this case I changed the condition into:
```java
if (!person.getVoided() && person.hasZeroNotVoidedNames()) {
   //mark validation error
}
```
The `person.hasZeroNotVoidedNames()` is as you've noticed is a little weird 😄
`hasNoVoidedNames` sounds like a person has only 'non-voided' names. But the logic is about having at least one 'non-voided' name, and any number of voided names. So I think `hasNoVoidedNames` would be a bit misleading.
`hasVoidedName` would not represent the logic inside.

I think this method should be called `hasNoNotVoidedNames` but that's a complex name 🤣 that's why I came up with `hasZeroNotVoidedNames`.

Reply:
How about? hasAtLeastOneNonVoidPersonName()